config/routes.rb

```diff
@@ -3,4 +3,12 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+
+  get "/dishes/:id", to: "dishes#show"
+  post "/dishes/:id", to: "dish_ingredient#create"
```

Comment:
A slightly more RESTful option for the path for this POST and DELETE would be `"/dishes/:id/ingredients"` but your controller organization is very RESTful!